src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -663,6 +663,16 @@ def set_argument_registry_server(self, registry_server):
     def set_argument_no_wait(self, no_wait):
         self.set_param("no_wait", no_wait)
 
+    def get_argument_enable_consumption_gpu(self):
+        return self.get_param("consumption_gpu_profile")
+
+    def validate_consumption_gpu_profile(self):
+        if self.get_argument_enable_consumption_gpu() is not None:
+            if self.get_argument_enable_consumption_gpu().lower() not in ["consumption-gpu-nc8as-t4", "consumption-gpu-nc4as-t4", "consumption-gpu-nc24-a100", "consumption-gpu-nc12-a100", "consumption-gpu-nv6ads-a10"]:
```

Comment:
Can you move this to the params?
`c.argument('consumption_gpu_profile', arg_type=get_enum_type(['consumption-gpu-nc8as-t4', 'consumption-gpu-nc4as-t4', 'consumption-gpu-nc24-a100'...`
Example:
https://github.com/Azure/azure-cli-extensions/blob/ede61b77d05e77dc06d9d82a2456f366dabd6638/src/containerapp/azext_containerapp/_params.py#L30